FEC-2765 Upgrade ExoPlayer to 2.11.3

giladna opened the pull request:
Support in Breaking changes
adding more player settings options
remove dummySurface test
remove duplicated Custom classes

giladna committed:
upgrade exo player

giladna committed:
cont

giladna committed:
cont

giladna committed:
cont

giladna committed:
cont

giladna committed:
Merge branch 'dev' into FEC_2765_01

giladna committed:
remove  redundant code

giladna committed:
lint

giladna committed:
drmSessionManager = new DefaultDrmSessionManager.Builder()
-                .setUuidAndExoMediaDrmProvider(MediaSupport.WIDEVINE_UUID, FrameworkMediaDrm.DEFAULT_PROVIDER)
-                .setMultiSession(true)
+                //.setUuidAndExoMediaDrmProvider(MediaSupport.WIDEVINE_UUID, FrameworkMediaDrm.DEFAULT_PROVIDER)
+                //.setMultiSession(true)
+                .setPlayClearSamplesWithoutKeys(allowClearLead)

giladna committed:
remove  DummySurfaceWorkaroundTest

giladna committed:
add more info
remove unneeded code since that is done internally now

giladna committed:
remove encoder report

giladna committed:
upgrade okHttp - 3.12.8 - no need to specify protocol

giladna committed:
add handleAudioBecomingNoisy to player settings

giladna committed:
refactor where we set the track selector parameter to the track selection helper
add set Max Video Size

giladna committed:
add new player settings

giladna committed:
cosmetics

giladna committed:
support new isLive logic

giladna committed:
add logging

giladna committed:
add extra information about decoder exceptions

giladna committed:
remove comment

giladna committed:
remove comment

giladna committed:
remove comment

giladna committed:
upgrade gradle 3.6.2

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/Player.java:
```diff
@@ -203,26 +205,89 @@
+         * @param maxVideoSizeSdEnabled -  Maximum allowed audio channel count.
+         * @return - Player Settings
+         */
+        Settings setMaxVideoSizeSd(boolean maxVideoSizeSdEnabled);
```
Remove, add comment to setMaxVideoSize()

noamtamim commented on playkit/src/main/java/com/kaltura/playkit/Player.java:
```diff
@@ -203,26 +205,89 @@
+         * @param preferredTextRoleFlags
+         * @return - Player Settings
+         */
+        Settings setPreferredTextRoleFlags(@C.RoleFlags int preferredTextRoleFlags);
```
Remove, for now.

giladna committed:
remoce CustomRendererFactory

giladna committed:
release drmSession only once -  so any other release call will not be executed

giladna committed:
remove unneeded setting

implement
 public void release() {
        if (realDrmSessionManager != null) {
            realDrmSessionManager.release();
        }
        if (realDrmSession != null) {
            realDrmSession.release();
        }
    }

giladna committed:
remove comment

giladna committed:
SessionWrapper remove

giladna committed:
no need to set to null

giladna committed:
.protocols(Collections.singletonList(Protocol.HTTP_1_1))    // Avoid http/2 due to https://github.com/google/ExoPlayer/issues/4078

giladna committed:
remove                         .setDrmSessionManager(DrmSessionManager.getDummyDrmSessionManager())

done in the constructor anyway

giladna committed:
catch Exception since the concrete exceptions might not exist

giladna committed:
catch Exception since the concrete exceptions might not exist
